client-spark/spark3/src/main/java/org/apache/spark/shuffle/DelegationRssShuffleManager.java

```diff
@@ -172,7 +181,78 @@ public <K, C> ShuffleReader<K, C> getReader(
       int endPartition,
       TaskContext context,
       ShuffleReadMetricsReporter metrics) {
-    return delegate.getReader(handle, startMapIndex, endMapIndex, startPartition, endPartition, context, metrics);
+    ShuffleReader<K, C> reader = null;
+    try {
+      // for spark 3.1.2
+      reader = (ShuffleReader<K, C>)delegate.getClass().getDeclaredMethod(
```

Comment:
why do we need reflection?

Reply:
Delegate  can also be SortShuffleManager.

Reply:
SortShuffleManager should also have this two methods.

Reply:
OK